### ISSUE TITLE
How do we handle exceeding free tier

### ISSUE THREAD

**Issue by mattttt** [raintank/worldping-api#61]
The worldPing free tier includes 3 endpoints (already covered in quota), however we currently do not have a plan for how to handle folks exceeding their predetermined # of Checks/mo. 

Short term, we've decided to handle this by proactively communicating to the user that they're over, but long term, we need an automated way to enforce a limit. 

Opening this issue to start the discussion.

**Comment by obfuscurity** [comment]
I don't know anything about how we handle billing, but wouldn't we just have different pricing tiers with an explicit allotment of checks (endpoints)?

**Comment by mattttt** [comment]
Currently, the only plan we're offering retail is Pay-as-you-go. 

Even for early access users that are on a plan (ex: 10M included for $35/mo) has an overage rate, without any hard limits. The free tier is the only one that would have a hard limit, since we do not have a credit card on file. 

I know you have mentioned from experience that people quite like pricing plans that do not allow for overages (consistent, month over month billing), stopping collecting new metrics instead when it hits that threshold. We just do not have the ability in the backend to do that yet to even consider it from a pricing perspective (though ill let @woodsaj chime in).

**Comment by woodsaj** [comment]
In order for the worldping-app and the backend to prevent users from configuration their endpoints to produce more the 3million checks in a month we would need to first know what plan they are on.  This would probably be best exposed when validating an APIKey.

Once that is in place we should be able to enforce limits.  It should be noted that the calculated ChecksPerMonth would be based on the assumption that all probes are online 100% of the time, which is not always true.   Public probes at least do have >99% uptime though.

**Comment by nopzor1200** [comment]
Do we think its necessary to have limits on the PayGo plan? Help me understand why? (In my mind PayGo plan is analogus to how Voxel used to do univeral transfer, or how AWS currently sells bandwidth)

If people want a flat rate they can do a custom plan (with custom commit). For custom plans we can elect not to autobill overages for customers that are concerned about that.

**Comment by mattttt** [comment]
Keep in mind, this issue is about restricting the free tier. The functionality happens to overlap with limiting a paid plan, should we choose to.
